src/common/ceph_time.h

```diff
@@ -499,6 +500,10 @@ namespace std {
 	std::chrono::duration_cast<timespan>(z) :
 	timespan(-z.count());
     }
+    inline timespan totimespan(signedspan z) {
+      assert(z >= signedspan::zero());
```

Comment:
Please use ceph_assert rather than assert. The assert mess is horrible and we're going to have to clean it up sometime, so we may as well make it easier.

Reply:
Sure.

Reply:
Oh, okay, allow me the off-topic for a bit here because this is relevant to my interests: are we just moving away from the nasty redefine of `assert` and use `ceph_assert` everywhere? Is there a timeline for that? Somewhere describing the reasoning, etc?

Reply:
WELL! I don't know of a timeline. It's something I want to do and I've heard other people talk about wanting to do, and the addition of things like ceph_assert and ceph_assert_always to assert.h suggests there's some plan somewhere.

The main /reasoning/ is that we keep fighting with every library we use and clobbering/unclobbering/reclobbering the definition and it's really obnoxious to have to reinclude include/assert.h.

Reply:
@adamemerson gotcha! thanks ;)